geomesa-hbase/geomesa-hbase-spark/src/main/scala/org/locationtech/geomesa/spark/hbase/HBaseSpatialRDDProvider.scala

```diff
@@ -0,0 +1,100 @@
+package org.locationtech.geomesa.spark.hbase
+
+import org.apache.commons.io.IOUtils
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.hbase.client.Scan
+import org.apache.hadoop.hbase.mapreduce.MultiTableInputFormat
+import org.apache.hadoop.hbase.protobuf.ProtobufUtil
+import org.apache.hadoop.hbase.util.Base64
+import org.apache.hadoop.io.Text
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.geotools.data.{DataStoreFinder, Query, Transaction}
+import org.locationtech.geomesa.hbase.data.{EmptyPlan, HBaseDataStore, HBaseDataStoreFactory, HBaseQueryPlan}
+import org.locationtech.geomesa.jobs.GeoMesaConfigurator
+import org.locationtech.geomesa.spark.SpatialRDDProvider
+import org.locationtech.geomesa.utils.geotools.FeatureUtils
+import org.opengis.feature.simple.SimpleFeature
+
+/**
+  * Created by afox on 2/8/17.
+  */
+class HBaseSpatialRDDProvider extends SpatialRDDProvider {
+  import org.locationtech.geomesa.spark.CaseInsensitiveMapFix._
+
+  override def canProcess(params: java.util.Map[String, java.io.Serializable]): Boolean =
+    HBaseDataStoreFactory.canProcess(params)
+
+  def rdd(conf: Configuration,
+          sc: SparkContext,
+          dsParams: Map[String, String],
+          origQuery: Query): RDD[SimpleFeature] = {
+    val ds = DataStoreFinder.getDataStore(dsParams).asInstanceOf[HBaseDataStore]
+
+    // get the query plan to set up the iterators, ranges, etc
+    lazy val sft = ds.getSchema(origQuery.getTypeName)
+    lazy val qp = ds.getQueryPlan(origQuery).head.asInstanceOf[HBaseQueryPlan]
+
+    if (ds == null || sft == null || qp.isInstanceOf[EmptyPlan]) {
+      sc.emptyRDD[SimpleFeature]
+    } else {
+      import org.locationtech.geomesa.index.conf.QueryHints._
+
+      val query = ds.queryPlanner.configureQuery(origQuery, sft)
+      GeoMesaConfigurator.setSerialization(conf)
+      query.getHints.getTransformSchema.foreach(GeoMesaConfigurator.setTransformSchema(conf, _))
+      GeoMesaConfigurator.setTable(conf, qp.table.getNameAsString)
+      GeoMesaConfigurator.setDataStoreInParams(conf, dsParams)
+      GeoMesaConfigurator.setFeatureType(conf, sft.getTypeName)
+      val scans = qp.ranges.map { s =>
+        val scan = s.asInstanceOf[Scan]
```

Comment:
these might sometimes be `Get`s - you might be able to use the common superclass `org.apache.hadoop.hbase.client.Query`